src/CommandingOfficers/Ave.java

```diff
@@ -542,33 +546,42 @@ public GameEventQueue getEvents(MapMaster gameMap)
 
     GlacioAbility(Ave commander)
     {
-      super(commander, GLACIO_NAME, GLACIO_COST);
+      super(GLACIO_NAME, GLACIO_COST);
       coCast = commander;
       damageMod = new CODamageModifier(GLACIO_BUFF);
       AIFlags = PHASE_TURN_END;
     }
 
     @Override
-    protected void perform(MapMaster gameMap)
+    protected void enqueueCOMods(Commander co, MapMaster gameMap, ArrayList<COModifier> modList)
     {
-      // Normal CO-power boost.
-      myCommander.addCOModifier(damageMod);
+      modList.add(damageMod);
+    }
 
+    @Override
+    protected void perform(Commander myCommander, MapMaster gameMap)
+    {
       // Freeze enemies around each of Ave's units or buildings.
-      for( Unit victim : findVictims(gameMap) )
+      for( Unit victim : findVictims(myCommander, gameMap) )
         victim.isStunned = true;
     }
 
     @Override
-    public GameEventQueue getEvents(MapMaster gameMap)
+    public GameEventQueue getEvents(Commander co, MapMaster gameMap)
     {
       // Keep track of any tiles that change to snow.
       ArrayList<MapChangeEvent.EnvironmentAssignment> tileChanges = new ArrayList<MapChangeEvent.EnvironmentAssignment>();
 
       // Add snow in an expanded range around Ave's areas.
-      int maxSnowRange = coCast.MAX_SNOW_SPREAD_RANGE + GLACIO_SNOW_SPREAD;
-      Set<XYCoord> tilesInRange = Utils.findLocationsNearProperties(gameMap, coCast, maxSnowRange);
-      tilesInRange.addAll(Utils.findLocationsNearUnits(gameMap, coCast, GLACIO_SNOW_SPREAD));
+      int maxSnowRange = GLACIO_SNOW_SPREAD;
+      if (co instanceof Ave)
+      {
+        maxSnowRange += ((Ave)co).MAX_SNOW_SPREAD_RANGE;
+      }
```

Comment:
Because the general future plan is to apply powers to all COs in a given `Army`.
It probably should be checking for equality with `coCast`, though, so you can't use two `Ave`s to expand your snow radius faster than normal.
(Or you could have `Nix`'s cost scale with the current radius explicitly, so the two `Ave`s would have their cost/range increase in sync but still benefit from the extra charge of a tag)

Reply:
As I understand it, we are no longer planning to apply powers to all COs in a given army?